models/glacier/avalanche_c/glacier_v2_avalanche_c_trades.sql

```diff
@@ -0,0 +1,102 @@
+{{ config(tags=['dunesql'],
+    alias = alias('trades')
+    ,partition_by = ['block_month']
+    ,materialized = 'incremental'
+    ,file_format = 'delta'
+    ,incremental_strategy = 'merge'
+    ,unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index']
+    ,post_hook='{{ expose_spells(\'["avalanche_c"]\',
+                                      "project",
+                                      "glacier",
+                                    \'["discochuck"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2023-03-30' %} -- min(evt_block_time) from glacier_avalanche_c.Pair_evt_Swap
+
+with dexs as (
+    -- glacier_exchange
+    SELECT t.evt_block_time                                             as block_time,
+           t.to                                                         as taker,
+           CAST(NULL as VARBINARY)                                      as maker,
+           case when amount0Out = UINT256 '0' then amount1Out else amount0Out end as token_bought_amount_raw,
+           case when amount0In = UINT256 '0' then amount1In else amount0In end    as token_sold_amount_raw,
+           null                                                          as amount_usd,
+           case when amount0Out = UINT256 '0' then f.token1 else f.token0 end     as token_bought_address,
+           case when amount0In = UINT256 '0' then f.token1 else f.token0 end      as token_sold_address,
+           t.contract_address                                           as project_contract_address,
+           t.evt_tx_hash                                                as tx_hash,
+           t.evt_index
+    FROM
+        {{ source('glacier_avalanche_c', 'Pair_evt_Swap') }} t
+    INNER JOIN {{ source('glacier_avalanche_c', 'PairFactory_evt_PairCreated') }} f
+        on f.pair = t.contract_address
+    {% if is_incremental() %}
+    WHERE {{ incremental_predicate('t.evt_block_time') }}
+    {% endif %}
+)
+select 'avalanche_c'                                             as blockchain,
+       'glacier'                                                 as project,
+       '1'                                                       as version,
```

Comment:
why is this spell named `v2`, yet it's set to `1` here?

Reply:
the schema file / seeds seem to reflect that too

Reply:
I will use version `2` here for now, as it seems like a uniswap_v2 forked contract.

Reply:
Yeah, sorry just got back from out of town. 

@Hosuke was right, it was due to it being a uni v2 fork, and the new CL pairs being based on uni v3 / algebra. I thought the v2/v3 was going to get muddled. Sort of a Vindicators 3 type scenario (rick and morty reference). 

Anyway, thank you both for all the help. I used quickswap as a base since they share the algebra v3-ish implementation and they seemed to have go through a bunch of v1/v2/v3 issues. I'm not well versed in all of this to figure out how to un muddle it.. or really to get it to work at all.

once again, thank you guys so much.